code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+        assertEquals(
+            Triple(
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX - 1,
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX,
+                PushTemplateConstants.DefaultValues.FILMSTRIP_CAROUSEL_CENTER_INDEX + 1
+            ),
+            result
+        )
+    }
+
+    @Test
+    fun `test populateManualCarouselImages with valid images`() {
+        val packageName = context.packageName
+        val centerIndex = 1
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        every { expandedLayout.addView(any(), any<RemoteViews>()) } just Runs
+        every { expandedLayout.setDisplayedChild(any(), any()) } just Runs
+
+        populateManualCarouselImages(
+            context,
+            pushTemplate.carouselItems,
+            packageName,
+            centerIndex,
+            pushTemplate,
+            trackerActivityClass,
+            expandedLayout
+        )
+
+        verify { expandedLayout.addView(any(), any<RemoteViews>()) }
+        verify { expandedLayout.setDisplayedChild(any(), centerIndex) }
+    }
+
+    @Test
+    fun `test populateManualCarouselImages with null images`() {
+        val packageName = context.packageName
+        val centerIndex = 1
+        every { getCachedImage(any()) } returns null
+
+        populateManualCarouselImages(
+            context,
+            pushTemplate.carouselItems,
+            packageName,
+            centerIndex,
+            pushTemplate,
+            trackerActivityClass,
+            expandedLayout
+        )
+
+        verify(exactly = 0) { expandedLayout.addView(any(), any<RemoteViews>()) }
+        verify(exactly = 0) { expandedLayout.setDisplayedChild(any(), centerIndex) }
+    }
+
+    @Test
+    fun `test populateFilmstripCarouselImages with valid images`() {
+        val mockBitmap = mockkClass(Bitmap::class)
+        val imageCaptions = listOf("Caption 1", "Caption 2", "Caption 3")
+        val imageClickActions = listOf("Action 1", "Action 2", "Action 3")
+        val newIndices = Triple(0, 1, 2)
+        every { getAssetCacheLocation() } answers { "any<String>()" }
+        every { getCachedImage(any()) } answers { mockBitmap }
+        every { expandedLayout.setTextViewText(any(), "Caption 2") } returns Unit
+        every { expandedLayout.setImageViewBitmap(any(), mockBitmap) } returns Unit
+        every {
+            expandedLayout.setRemoteViewClickAction(
+                context,
+                trackerActivityClass,
+                any(),
+                any(),
+                null,
+                any()
+            )
+        } returns Unit
+        populateFilmstripCarouselImages(
+            context,
+            imageCaptions,
+            imageClickActions,
+            newIndices,
+            pushTemplate,
+            trackerActivityClass,
+            expandedLayout
+        )
+        verify(exactly = 1) {
```

Comment:
Can you also verify `expandedLayout.setImageViewBitmap` is called `exactly=3` with the correct values?